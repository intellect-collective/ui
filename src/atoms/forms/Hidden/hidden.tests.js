import React from 'react';
import Hidden from '.';
import {
    expectConsoleError,
    getMissingFieldError,
    rerenderSuppressionTests
} from '../../../../test/utils';

describe('Hidden', () => {
    rerenderSuppressionTests(Hidden, { name: 'test' });

    it('Should throw an error on missing name', () => {
        expectConsoleError(() => {
            mount(<Hidden />);
        }, getMissingFieldError('name'));
    });

    it('Should accept valid props', () => {
        const wrapper = mount(<Hidden name="test" value="sausages" />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('input').prop('name')).toEqual('test');
        expect(wrapper.find('input').prop('value')).toEqual('sausages');
    });
});
