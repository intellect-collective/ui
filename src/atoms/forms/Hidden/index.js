import React from 'react';
import formContext from '../_decorators/formContext';

const Hidden = (props) => (
    <input { ...props } type="hidden" />
);
Hidden.displayName = 'Hidden';

export default formContext()(Hidden);
