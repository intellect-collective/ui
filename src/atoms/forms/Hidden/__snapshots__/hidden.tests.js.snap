// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hidden Should accept valid props 1`] = `
<FormContext
  name="test"
  value="sausages"
>
  <Hidden
    name="test"
    value="sausages"
  >
    <input
      name="test"
      type="hidden"
      value="sausages"
    />
  </Hidden>
</FormContext>
`;
